packages/core/src/onboarding.ts

```diff
@@ -153,9 +155,22 @@ export const setOnboardingStage = (stage: Partial<IOnboardingStage>) => {
         tempStage.iconClass = stage.iconClass
           ? stage.iconClass
           : tempStage.iconClass
+        newStage = tempStage
         break
       }
     }
+    const tempMessage = get(onboardingMessages).filter(m => m.onboardingStage.id === stage.id)
+    const updateMsg = tempMessage.map((m) => {
+      return {
+        ...m,
+        onboardingStage: newStage
+      }
+    })
+
+    onboardingMessages.set(updateMsg)
```

Comment:
As you filtered the onboarding messages on top, you are now deleting all messages form other stages when you only set the updateMsg here